tasks/src/main/scala/dagr/tasks/picard/CollectIlluminaLaneMetrics.scala

```diff
@@ -29,6 +29,23 @@ import dagr.tasks.DagrDef.DirPath
 
 import scala.collection.mutable.ListBuffer
 
+object CollectIlluminaLaneMetrics {
+  def convertBarcodeAndSkipToTemplate(readStructure: String): String = {
+    readStructure
```

Comment:
I'm finding this super hard to follow - but there are tests so if you're happy I'm happy.

I think if I were to want to understand this in the future I would break it up a bit and give the intermediates really clear names and types so I could see what's going on.